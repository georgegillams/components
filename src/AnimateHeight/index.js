import AnimateHeight from './AnimateHeight';

export default AnimateHeight;
export { AnimateHeight };
