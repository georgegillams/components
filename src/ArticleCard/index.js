import ArticleCard, { ARTICLE_CARD_LAYOUTS } from './ArticleCard';

export default ArticleCard;
export { ArticleCard, ARTICLE_CARD_LAYOUTS };
