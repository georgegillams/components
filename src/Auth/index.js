import AdminOnly from './AdminOnly';
import APIEntity from './APIEntity';
import ObjectAsList from './ObjectAsList';
import DebugObject from './DebugObject';
import EmailVerifiedOnly from './EmailVerifiedOnly';
import LoadingCover from './LoadingCover';
import LoggedInOnly from './LoggedInOnly';
import LoggedOutOnly from './LoggedOutOnly';
import Status from './Status';
import User from './User';

export default {
  APIEntity,
  AdminOnly,
  DebugObject,
  LoadingCover,
  EmailVerifiedOnly,
  LoggedInOnly,
  LoggedOutOnly,
  ObjectAsList,
  Status,
  User,
};
export {
  APIEntity,
  AdminOnly,
  DebugObject,
  LoadingCover,
  EmailVerifiedOnly,
  LoggedInOnly,
  LoggedOutOnly,
  ObjectAsList,
  Status,
  User,
};
