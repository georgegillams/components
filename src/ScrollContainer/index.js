import withScroll from './withScroll';

export default withScroll;
export { withScroll };
