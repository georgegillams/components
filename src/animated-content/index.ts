export { default, ANIMATIONS } from './animated-content';
