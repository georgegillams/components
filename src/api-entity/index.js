export { default } from './api-entity';
