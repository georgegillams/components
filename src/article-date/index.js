export { default } from './article-date';
