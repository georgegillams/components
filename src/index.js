import {
  APIEntity,
  AdminOnly,
  DebugObject,
  LoadingCover,
  EmailVerifiedOnly,
  LoggedInOnly,
  LoggedOutOnly,
  ObjectAsList,
  Status,
  User,
} from './Auth';
import { Card, ArticleCard, ARTICLE_CARD_LAYOUTS } from './Cards';
import { Code, CodeInline, CodeBashArrow } from './Code';
import { CreativeCommons } from './CreativeCommons';
import { DegreeModule } from './Degree';
import { ColourPalleteItem } from './Design';
import { Footer } from './Footer';
import { Button, BurgerButton } from './Button';
import { LoadingIndicator } from './LoadingIndicator';
import { Logo } from './Logo';
import { MoneyPot } from './MoneyPot';
import { NavigationBar, NavigationItem } from './NavigationBar';
import {
  NotificationCollection,
  NOTIFICATION_TYPES,
  NotificationComp,
} from './Notifications';
import { Progress } from './Progress';
import { GGRedirect } from './Redirect';
import { RequestStatusContainer, RequestStatus } from './RequestStatus';
import {
  Skeleton,
  ButtonSkeleton,
  CardSkeleton,
  CheckboxSkeleton,
  CompactCardSkeleton,
  InputSkeleton,
  NotificationSkeleton,
  ProgressSkeleton,
  SectionSkeleton,
  SmallButtonSkeleton,
  SmallProgressSkeleton,
  SubSectionSkeleton,
  TextLinkSkeleton,
  TicketStatusSkeleton,
} from './Skeletons';
import { withScroll } from './ScrollContainer';
import { Tag, TagFilter, TAG_TYPES } from './Tag';
import {
  ArticleDate,
  Quote,
  Section,
  SubSection,
  MarkdownRenderer,
  AnimatedContent,
  TextLink,
} from './Typography';

export {
  APIEntity,
  ARTICLE_CARD_LAYOUTS,
  AdminOnly,
  AnimatedContent,
  ArticleCard,
  ArticleDate,
  BurgerButton,
  Button,
  ButtonSkeleton,
  Card,
  CardSkeleton,
  CheckboxSkeleton,
  Code,
  CodeBashArrow,
  CodeInline,
  ColourPalleteItem,
  CompactCardSkeleton,
  CreativeCommons,
  DebugObject,
  DegreeModule,
  EmailVerifiedOnly,
  Footer,
  GGRedirect,
  InputSkeleton,
  LoadingCover,
  LoadingIndicator,
  LoggedInOnly,
  LoggedOutOnly,
  Logo,
  MarkdownRenderer,
  MoneyPot,
  NOTIFICATION_TYPES,
  NavigationBar,
  NavigationItem,
  NotificationCollection,
  NotificationComp,
  NotificationSkeleton,
  ObjectAsList,
  Progress,
  ProgressSkeleton,
  Quote,
  RequestStatus,
  RequestStatusContainer,
  Section,
  SectionSkeleton,
  Skeleton,
  SmallButtonSkeleton,
  SmallProgressSkeleton,
  Status,
  SubSection,
  SubSectionSkeleton,
  TAG_TYPES,
  Tag,
  TagFilter,
  TextLink,
  TextLinkSkeleton,
  TicketStatusSkeleton,
  User,
  withScroll,
};
