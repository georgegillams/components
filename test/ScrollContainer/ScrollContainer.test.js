import { cleanRestScrollProps } from '../../src/ScrollContainer';

describe('ScrollContainer', () => {
  it('should clean rest props correctly', () => {
    const rest = {
      fullyInView: false,
      hasBeenFullyInView: false,
      hasBeenInView: false,
      hasBeenJustInView: false,
      hasBeenMostlyInView: false,
      hasBeenOutOfView: false,
      inView: false,
      justInView: false,
      mostlyInView: false,
      outOfView: false,
      scrollPosition: 12,
      scrollPositionVh: 10,
      nonClearableProp: 'asdf',
    };
    cleanRestScrollProps(rest);
    expect(rest).toMatchSnapshot();
  });
});
