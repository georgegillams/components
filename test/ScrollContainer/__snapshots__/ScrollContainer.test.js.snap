// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ScrollContainer should clean rest props correctly 1`] = `
Object {
  "nonClearableProp": "asdf",
}
`;
